ubi_manifest/worker/tasks/depsolver/utils.py

```diff
@@ -162,6 +171,20 @@ def __ne__(self, other):
     return Klass
 
 
+def is_requirement_resolved(requirement, provider):
+    if requirement.flags:
+        req_evr = (requirement.epoch, requirement.version, requirement.release)
+        prov_evr = (provider.epoch, provider.version, provider.release)
+        # compare provider with requirement
+        out = RELATION_CMP_MAP[requirement.flags](prov_evr, req_evr)
```

Comment:
I'm a bit worried that this approach may be too naive. Even the semantic versioning allow for some strange combinations which are not comparable alphabetically, see https://semver.org/#spec-item-11 . On the other hand, the versions we use are mostly numeric. Even though we do have some outliers here and there, all of those I have been able to find are comparable also alphabetically with the expected outcome, so treat this comment only as a warning.

Reply:
well, it uses `labelCompare` function from RPM python bindings so any valid EVR combination for RPM naming should  be fine.